client/src/pages/admin/AdminLibraryUpdateForm.tsx

```diff
@@ -0,0 +1,270 @@
+    [onChange]
+  );
+
+  return (
+    <Box>
+      <Typography variant="h5" textAlign={'left'}>
+        {dayName}
+      </Typography>
+      <Stack
+        alignItems={'center'}
+        sx={{ width: '100%', mb: 2, minHeight: '60px' }}
+        direction="row"
+        spacing={2}
+        divider={<Divider orientation="vertical" flexItem />}>
+        <Stack direction="row" alignItems="center">
+          <Checkbox checked={open} onChange={() => handleOpenChange(!open)} />
+          <Typography onClick={() => handleOpenChange(!open)} sx={{ cursor: 'pointer' }}>
+            Otwarte?
+          </Typography>
+        </Stack>
+        {open && (
+          <>
+            <TimeInputField label="Otwarcie" field={fields[0]} control={control} disabled={!open} />
+            <TimeInputField label="Zamknięcie" field={fields[1]} control={control} disabled={!open} />
+          </>
+        )}
+      </Stack>
+    </Box>
+  );
+}
+
+type DayOfWeek = {
+  dayName: string;
+  openTimeField: keyof AddLibraryType;
+  closeTimeField: keyof AddLibraryType;
+};
+
+const daysOfWeek: DayOfWeek[] = [
+  {
+    dayName: 'Poniedziałek',
+    openTimeField: 'mondayOpenTime',
+    closeTimeField: 'mondayCloseTime',
+  },
+  {
+    dayName: 'Wtorek',
+    openTimeField: 'tuesdayOpenTime',
+    closeTimeField: 'tuesdayCloseTime',
+  },
+  {
+    dayName: 'Środa',
+    openTimeField: 'wednesdayOpenTime',
+    closeTimeField: 'wednesdayCloseTime',
+  },
+  {
+    dayName: 'Czwartek',
+    openTimeField: 'thursdayOpenTime',
+    closeTimeField: 'thursdayCloseTime',
+  },
+  {
+    dayName: 'Piątek',
+    openTimeField: 'fridayOpenTime',
+    closeTimeField: 'fridayCloseTime',
+  },
+  {
+    dayName: 'Sobota',
+    openTimeField: 'saturdayOpenTime',
+    closeTimeField: 'saturdayCloseTime',
+  },
+  {
+    dayName: 'Niedziela',
+    openTimeField: 'sundayOpenTime',
+    closeTimeField: 'sundayCloseTime',
+  },
+];
+
+function AdminLibraryUpdateForm() {
+  const navigate = useNavigate();
+  const params = useParams();
+
+  const { data, status } = useQuery({
+    queryKey: ['libraries', params.libraryId],
+    queryFn: () => getLibrary(params.libraryId + ''),
+  });
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+    unregister,
+  } = useForm<AddLibraryType>({
+    resolver: zodResolver(AddLibrary)
+  });
+
+  const updateLibraryMutation = useMutation({
+    mutationFn: updateLibrary,
+    onSuccess: () => {
+      navigate('..');
+    },
+    onError: (e: Error) => {
+      console.error(e);
+    },
+  });
+
+  const deleteLibraryMutation = useMutation({
+    mutationFn: deleteLibrary,
+    onSuccess: () => {
+      navigate('..');
+    },
+    onError: (e: Error) => {
+      console.error(e);
+    },
+  });
+
+  const onSubmit = useCallback(
+    async (data: AddLibraryType) => {
+      updateLibraryMutation.mutate({ library: data, id: params.libraryId! });
+    },
+    [params.publisherId, updateLibraryMutation]
+  );
```

Comment:
w sumie to nie ma potrzeby żeby to było w useCallback chyba, no i nie musiby być `async`